src/generate-simple-manifest.js

```diff
@@ -0,0 +1,51 @@
+const { validateGeneratedManifest } = require('./validate-generated-manifest.js')
+
+const generateSimpleManifest = function (manifest) {
+  // This function generates a simplified manifest with the container
+  // environment fields fully interpolated
+
+  const manifestErrors = validateGeneratedManifest(manifest)
+  if (manifestErrors.length) {
+    throw new Error(`Invalid manifest. errors=${JSON.stringify(manifestErrors, null, 2)}`)
+  }
+  const simpleManifest = {
+    manifest: {
+      name: manifest['manifest']['name'],
+      version: manifest['manifest']['version'],
+      machine: manifest['manifest']['machine'],
+      containers: manifest['manifest']['containers']
+    }
+  }
+
+  const simpleContainers = simpleManifest['manifest']['containers']
+  simpleContainers.map((container) => {
+    processContainer(manifest, container)
+  })
+  return simpleManifest
+}
+
+const processContainer = function (manifest, container) {
+  const environment = container['environment']
```

Comment:
Note there are some checks on the environment fields that are left out. Those checks are already performed when the manifest is validated.